src/compiler/sys.ts

```diff
@@ -1507,7 +1501,7 @@ export let sys: System = (() => {
             getAccessibleSortedChildDirectories: path => getAccessibleFileSystemEntries(path).directories,
             realpath,
             tscWatchFile: process.env.TSC_WATCHFILE,
-            useNonPollingWatchers: process.env.TSC_NONPOLLING_WATCHER,
+            useNonPollingWatchers: !!process.env.TSC_NONPOLLING_WATCHER,

```

Comment:
This one is really suspicious; should we be parsing this rather than checking for truthiness? I made it `!!` to match current behavior, but, it seems unexpected.

Reply:
@sheetalkamat Do you have any advice for this particular place? Is this "fine" or was it intended that we let you set `TSC_NONPOLLING_WATCHER=false` or `TSC_NONPOLLING_WATCHER=0` or `TSC_NONPOLLING_WATCHER=off` or similar?

Reply:
I think its fine to fix it this way. This is from way back... given we have watchOptions now i dont think it matters. We probably should deprecate these environment variable options.

Reply:
Great, thanks!

The main benefit of the environment variables is being able to set them without modifying tsconfig, which is helpful if you're on a system where watch mode is broken and you don't want to commit a tsconfig change. (e.g. #52876 #52535)